x-pack/plugins/observability/public/components/shared/exploratory_view/lens_embeddable.tsx

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import React, { Dispatch, SetStateAction, useCallback } from 'react';
+import { combineTimeRanges } from './exploratory_view';
+import { TypedLensByValueInput } from '../../../../../lens/public';
+import { useSeriesStorage } from './hooks/use_series_storage';
+import { ObservabilityPublicPluginsStart } from '../../../plugin';
+import { useKibana } from '../../../../../../../src/plugins/kibana_react/public';
+
+interface Props {
+  lensAttributes: TypedLensByValueInput['attributes'];
+  setLastUpdated: Dispatch<SetStateAction<number | undefined>>;
+}
+
+export function LensEmbeddable(props: Props) {
+  const { lensAttributes, setLastUpdated } = props;
+
+  const {
+    services: { lens, notifications },
+  } = useKibana<ObservabilityPublicPluginsStart>();
+
+  const LensComponent = lens?.EmbeddableComponent;
+
+  const { firstSeriesId, firstSeries: series, setSeries, allSeries } = useSeriesStorage();
+
+  const timeRange = combineTimeRanges(allSeries, series);
```

Comment:
Here's one more area of optimization I see. This variable will cause `LensComponent` to rerender on every parent rerender. We can prevent this with `useMemo`, and it may be worth it for this heavy canvas based component.

Reply:
yes i think that totally makes sense, i will do the change